app/views/do.py

```diff
@@ -2363,3 +2391,96 @@ def close_comment_report(id, action):
 
     else:
         return jsonify(status='error', error=_('Failed to update report'))
+
+
+@do.route('/do/report/close_post_related_reports/<related_reports>', methods=['POST'])
+@login_required
+def close_post_related_reports(related_reports):
+    related_reports = json.loads(related_reports)
+    error = ''
+    # ensure user is mod or admin and report, post, and sub exist
+    for related_report in related_reports:
+        try:
+            report = SubPostReport.get(SubPostReport.id == related_report['id'])
+        except SubPostReport.DoesNotExist:
+            error = jsonify(status='error', error=_('Report does not exist'))
+
+        try:
+            post = SubPost.get(SubPost.pid == report.pid)
+        except SubPost.DoesNotExist:
+            error = jsonify(status='error', error=_('Post does not exist'))
+
+        try:
+            sub = Sub.get(Sub.sid == post.sid)
+        except Sub.DoesNotExist:
+            error = jsonify(status='error', error=_('Sub does not exist'))
+
+        if not current_user.is_mod(sub.sid) and not current_user.is_admin():
+            error = jsonify(status='error', error=[_('Not authorized')])
```

Comment:
You're returning string errors in this function except in this line where it's an array